torch/csrc/jit/codegen/cuda/graph_fuser.cpp

```diff
@@ -2051,12 +2051,14 @@ void decomposeLinearOps(Block* block) {
     auto mat1_size =
         n->input(1)->type()->cast<c10::TensorType>()->sizes().concrete_sizes();
 
-    // TODO: The assert is not necessary when we can handle matmul, right now we
-    // are splitting the linear between matmul & bias_add. Our fuser can only
-    // take the second half and we would need the size information.
-    TORCH_INTERNAL_ASSERT(
-        mat0_size.has_value() && mat1_size.has_value(),
```

Comment:
@csarofeen we have aggressive assert here on profiling sizes being static and available, which is not the case in the python example, when the profiled section is in the loop and sees shape changes during profile runs.

So the updated code is just to skip the decomposition and not fusing it.
Technically speaking, we can probably relax the check here a little bit and still handle decomposition with the profiled dynamic shape, but I don't see it as a pressing issue at the time and would like to keep the coder simpler.